File: frontend/src/types/anomaly.ts
export interface AnomalyRecord {
  anomaly_id: string;
  camera_id: string;
  seat_model_id: string | null;
  region_id: string | null;
  source: string;
  anomaly_score: number | null;
  date_folder: string;
  status: string;
  detected_at: string;
  original_url: string | null;
  heatmap_url: string | null;
  crop_url: string | null;
  crop_urls: string[];
  cluster_id: string | null;
  created_at: string;
  trace_id: string | null;
}

/** 轻量 anomaly 摘要，用于 cluster 详情页列表展示。 */
export interface AnomalySummary {
  anomaly_id: string;
  camera_id: string;
  seat_model_id?: string | null;
  region_id?: string | null;
  anomaly_score: number | null;
  status: string;
  crop_url: string | null;
  detected_at: string | null;
}

/** POST /api/anomaly/{id}/reprocess 响应 */
export interface AnomalyReprocessResponse {
  status: string;
  anomaly_id: string;
}

/** DELETE /api/anomaly/{id} 响应 */
export interface AnomalyDeleteResponse {
  status: string;
  anomaly_id: string;
}

/** GET /api/embedding/search 响应 - 相似 embedding 搜索结果 */
export interface EmbeddingSimilarResult {
  anomaly_id: string;
  similarity: number;
  seat_model_id?: string | null;
  camera_id?: string | null;
  region_id?: string | null;
  date_folder?: string | null;
  crop_url?: string | null;
}

File: backend/app/schemas/anomaly.py
from __future__ import annotations

from datetime import datetime
from typing import Optional

from pydantic import BaseModel, Field

from app.schemas import CURRENT_SCHEMA_VERSION


class AnomalyUploadResponse(BaseModel):
    anomaly_ids: list[str]
    count: int
    status: str = "received"
    message: str = "Anomaly queued for processing"
    schema_version: str = CURRENT_SCHEMA_VERSION


class AnomalyQueryParams(BaseModel):
    camera_id: str | None = None
    seat_model_id: str | None = None
    region_id: str | None = None
    source: str | None = None
    status: str | None = None
    date_from: str | None = None
    date_to: str | None = None
    min_score: float | None = Field(default=None, ge=0.0)
    max_score: float | None = Field(default=None, ge=0.0)
    page: int = Field(default=1, ge=1)
    page_size: int = Field(default=20, ge=1, le=100)


class AnomalyResponse(BaseModel):
    anomaly_id: str
    camera_id: str
    seat_model_id: str | None = None
    region_id: str | None = None
    source: str
    anomaly_score: float | None
    date_folder: str
    status: str
    detected_at: datetime
    original_url: str | None = None
    heatmap_url: str | None = None
    crop_url: str | None = None
    crop_urls: list[str] = []
    proposal_count: int = 0
    proposals_json: Optional[str] = None
    cluster_id: str | None = None
    created_at: datetime
    trace_id: str | None = None

    model_config = {"from_attributes": True}
